server/src/main/java/org/elasticsearch/health/components/controller/ClusterCoordination.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health.components.controller;
+
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.health.HealthComponent;
+import org.elasticsearch.health.HealthIndicator;
+import org.elasticsearch.health.HealthStatus;
+
+import java.util.Collections;
+
+public final class ClusterCoordination {
+
+    public static final String NAME = "cluster_coordination";
+
+    public static final String INSTANCE_HAS_MASTER_NAME = "instance_has_master";
+    public static final String INSTANCE_HAS_MASTER_GREEN_SUMMARY = "Health coordinating instance has a master node.";
+    public static final String INSTANCE_HAS_MASTER_RED_SUMMARY = "Health coordinating instance does not have a master node.";
+
+    private ClusterCoordination() {}
+
+    public static HealthComponent createClusterCoordinationComponent(
```

Comment:
I think we should turn this around and have a `HasMasterHealthIndicatorService` that can calculate the health for an indicator and then gather the components out of it.

We could remove this from this initial PR to unblock the merge of it.

Reply:
I can do this in a follow-up.